geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/RecordIngestProcessor.scala

```diff
@@ -79,8 +79,14 @@ trait RecordIngestProcessor extends DataStoreIngestProcessor {
         override def process(in: InputStream): Unit = {
           WithClose(recordReaderFactory.createRecordReader(file, in, logger)) { reader =>
             val converter = SimpleFeatureRecordConverter(reader.getSchema, opts)
+            val userData = loadFeatureTypeUserData(converter.sft, context, file)
+            val sft = if (userData.isEmpty) { converter.sft } else {
+              val copy = SimpleFeatureTypes.copy(converter.sft)
+              copy.getUserData.putAll(userData)
+              copy
+            }
```

Comment:
It seems like `loadFeatureTypeUserData` could be changed to return an SFT with the userData set from the converter, context, and file. 

If that makes sense, that'd avoid repeating this block of code in two places.  A small consideration.

Reply:
My suggestion may be tantamount to renaming the method `decorate` and widening its inputs to include the context and flowfile.

Reply:
yeah that was the original way it was done, but i thought it was maybe less prone to errors to have the feature type method not cause side effects. there are some places that we cache the feature types, and i didn't want to cache the modified ones. is that a good enough justification? i'm not too opposed to changing it if you feel strongly...